### PR TITLE
[ISSUE-457][BUG] After worker recover, we should make sure there is no partial sortedFile and indexFile when new FileSorter task started

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?

If during shutdown worker, FileSorter task is interrupted and remain partial sorted file under path, after restart worker, same FileSorter task may write corrupted sortedFile.
We should make sure when starting sorted file, no partial file remain.

### Related issues.
https://github.com/alibaba/RemoteShuffleService/issues/457

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture @FMX

File: server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java [hunk 1]
```diff
@@ -415,24 +414,30 @@ static class ShuffleBlockInfo {
...
     FileSorter(File originFile, long originFileLen, String fileId, String shuffleKey) {
       this.originFile = originFile;
-      this.sortedFileName = originFile.getAbsolutePath() + SORTED_SUFFIX;
-      this.indexFileName = originFile.getAbsolutePath() + INDEX_SUFFIX;
+      this.sortedFile = new File(originFile.getParentFile(), originFile.getName() + SORTED_SUFFIX);
+      this.indexFile = new File(originFile.getParentFile(), originFile.getName() + INDEX_SUFFIX);
       this.originFileLen = originFileLen;
       this.fileId = fileId;
       this.shuffleKey = shuffleKey;
+      if (sortedFile.exists()) {
```
Comment (FMX) [inline server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java:430]: The last step of sorting a file is to delete the origin file. So I think the condition can be changed to check whether the original file exists or not. If the origin file does not exist, that means sort probably succeeds.

Comment (AngersZhuuuu) [reply server-worker/src/main/java/com/aliyun/emr/rss/service/deploy/worker/storage/PartitionFilesSorter.java:430]: > The last step of sorting a file is to delete the origin file. So I think the condition can be changed to check whether the original file exists or not. If the origin file does not exist, that means sort probably succeeds.

For this pr, I am considering the case that originFile and sortedFiles both exist, and the `sortedFiles` map recovered from LevelDB not contain this fils's information. Then new `openStream` request come,  then we will write to a existed sortedFile.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**